assets/javascripts/bonus_time.js

```diff
@@ -91,6 +91,18 @@ bonus_time.redmine_date = function(date) {
   return date.getFullYear() + '-' + month + '-' + date.getDate();
 }
 
+bonus_time.check_numeric_hours = function(e) {
+  var $input = $('#time_entry_hours');
+  var hours = parseFloat($input.val());
+  if (NaN == hours || 0 == hours) {
+    $input.val('');
+    alert('Please enter a valid number of hours');
+    e.preventDefault();
```

Comment:
not a big deal, but you shouldn't need to `preventDefault()` if you're returning false